spec/controllers/items_controller_spec.rb

```diff
@@ -34,46 +33,73 @@
   end
 
   context "new item is created successfully" do
+    before(:each) do
+      user =  User.create(username: "dreamteam", email: "us@dream.com", password_hash: "zzzzzzz")
+      session[:user_id] = user.id
+    end
+
     it "response is successful" do
-      post :create, :item => {name: "a", description: "A", price: 1, email: "a"}
+      post :create, :item => {name: "a", description: "A", price: 1, user_id: User.last.id}
 
       expect(response).to have_http_status(:redirect)
       expect(response.content_type).to eq("text/html")
     end
 
-    it "redirects to the root" do
-      post :create, :item => {name: "a", description: "A", price: 1, email: "a"}
+    it "redirects to items index" do
+      post :create, :item => {name: "a", description: "some description", price: 1, user_id: User.last.id}
 
-      expect(response).to redirect_to(:root)
+      expect(response).to redirect_to items_path
     end
 
     it "adds the item to the database" do
-      post :create, :item => {name: "ping-pong paddle", description: "you beat Alex with it", price: 1, email: "a"}
+      post :create, :item => {name: "ping-pong paddle", description: "you beat Alex with it", price: 1, user_id: User.last.id}
 
       expect(Item.first.name).to eq("ping-pong paddle")
       expect(Item.count).to eq(1)
     end
   end
 
   context "new item is invalid and not created" do
+    before(:each) do
+      user =  User.create(username: "dreamteam", email: "us@dream.com", password_hash: "zzzzzzz")
+      session[:user_id] = user.id
+    end
+
     it "response is successful" do
-      post :create, :item => {name: "", description: "A", price: 1, email: "a"}
+      post :create, :item => {name: "", description: "some description", price: 1, user_id: User.last.id}
 
       expect(response).to have_http_status(:success)
       expect(response.content_type).to eq("text/html")
     end
 
     it "renders the new template" do
-      post :create, :item => {name: "", description: "A", price: 1, email: "a"}
+      post :create, :item => {name: "", description: "A", price: 1, user_id: User.last.id}
 
       expect(response).to render_template(:new)
     end
 
     it "does not save the item to the database" do
-      post :create, :item => {name: "", description: "A", price: 1, email: "a"}
+      post :create, :item => {name: "", description: "A", price: 1, user_id: User.last.id}
 
       expect(Item.count).to eq(0)
     end
   end
-end
 
+  context "no current_user" do
+    before(:each) do
+      user =  User.create(username: "dreamteam", email: "us@dream.com", password_hash: "zzzzzzz")
```

Comment:
Again, this could be a let. Your test setup should be clearer, as described in the above comment.